scripts/mktest.py

```diff
@@ -0,0 +1,277 @@
+import requests
+import xml.etree.ElementTree as ET
+import tilequeue.tile as tile
+from ModestMaps.Core import Coordinate
+from jinja2 import Template
+import os
+from contextlib import contextmanager
+
+
+def way_full(way_id, zoom=16):
+    url = 'https://api.openstreetmap.org/api/0.6/way/%d/full' % (way_id,)
+    headers = {'user-agent': 'mktest.py/0.0.1 (https://github.com/tilezen)'}
+    r = requests.get(url, headers=headers)
+    root = ET.fromstring(r.content)
+    assert root.tag == 'osm'
+
+    nodes = {}
+    nds = []
+    tags = {}
+
+    for child in root:
+        if child.tag == 'node':
+            node_id = int(child.attrib['id'])
+            nodes[node_id] = child
+
+        elif child.tag == 'way':
+            assert way_id == int(child.attrib['id'])
+            for wchild in child:
+                if wchild.tag == 'nd':
+                    nds.append(int(wchild.attrib['ref']))
+                elif wchild.tag == 'tag':
+                    k = wchild.attrib['k']
+                    v = wchild.attrib['v']
+                    tags[k] = v
+
+    assert nds
+    first_node = nodes[nds[0]]
+    lat = float(first_node.attrib['lat'])
+    lon = float(first_node.attrib['lon'])
+    x, y = tile.deg2num(lat, lon, zoom)
```

Comment:
And we only care about the first one because we're interested in tags here, and not the geometry? (Just making sure I understand).

Reply:
Yup. I was thinking that, because some of the tests are mildly geometry-dependent with the country code tests, I wanted to have some representative geometry but not the whole thing. So I figured it would probably work out okay to find the tile containing any node in the way and use the `dsl.tile_diagonal` function to fake up some geometry in a concise statement.

That works great for the roads, but we might want to do something else as we add extra test generators - sometimes using the real geometry when it's important, sometimes generating it when it isn't.